BeagleBone_black/src/navigation/navigation.cpp

```diff
@@ -75,6 +75,14 @@ void Navigation::update(ImuArray imus)
   this->accelerometerUpdate(DataPoint<NavigationVector>(imus[0].acc.timestamp, avg));
 }
 
+std::array<NavigationType, 3> Navigation::getNearestStripeDists()
+{
+  std::array<NavigationType, 3> arr;
+  for (unsigned int i = 0; arr.size(); ++i)
+    arr[i] = kStripeLocations[stripe_count_ + i] - getDisplacement();
```

Comment:
you should somehow offset the `kStripeLocations`, for example, if `stipe_count_==0` (at the start of the run) you would calculate that last stripe was at +30.48, whereas it should be something negative (or at most 0)

also, are you not worried about the array bounds? what if we have `stipe_count_ > 41` ?